contracts/os/andromeda-kernel/src/execute.rs

```diff
@@ -58,16 +58,28 @@ pub fn amp_receive(
             error: Some("No messages supplied".to_string())
         }
     );
+
+    let mut res_messages = Vec::new();
+    let mut res_attributes = Vec::new();
+    let mut res_events = Vec::new();
+
     for (idx, message) in packet.messages.iter().enumerate() {
         let mut handler = MsgHandler::new(message.clone());
-        res = handler.handle(
+        let msg_res = handler.handle(
             deps.branch(),
             info.clone(),
             env.clone(),
             Some(packet.clone()),
             idx as u64,
         )?;
+        res_messages.extend_from_slice(&msg_res.messages);
```

Comment:
It might be best to just extend the `res` object directly here instead of making the temporary `res_*` variables. Unless there's a specific reason?